raiden/raiden_service.py

```diff
@@ -338,6 +360,9 @@ def handle_state_change(self, state_change, block_number=None):
 
         event_list = self.wal.log_and_dispatch(state_change, block_number)
 
+        if not self.dispatch_events:
```

Comment:
why not use a lock here instead?

Reply:
how would you use a lock to switch the logic from ignoring the events to applying the events?

Reply:
```
def start_async():
  ...
  self.disable_dispatch = Semaphore(1)
  with self.disable_dispatch:
    self.dispatch_pending()

def handle_state_change(self, ...):
   ...
   if self.disable_dispatch.locked() is True:
     return []
```

As a side note, nothing prevents public `start_async()` from being run twice.

Reply:
@pcppcp i would imagine this can happen if you launch Raiden in `--console` and manage to call `start_async`.  I added the lock mechanism you've described.